### PR TITLE
Get rid of externs/olx.js and src/ol/typedefs.js typedefs for ol/ and ol/events/

### PR DESCRIPTION
**Pull request by ahocevar**
See #7947.

### CODE REVIEW

File: src/ol/Feature.js [hunk 1]
```diff
@@ -231,9 +236,9 @@ Feature.prototype.setGeometry = function(geometry) {
  * Set the style for the feature.  This can be a single style object, an array
  * of styles, or a function that takes a resolution and returns an array of
  * styles. If it is `null` the feature has no style (a `null` style).
- * @param {ol.style.Style|Array.<ol.style.Style>|ol.StyleFunction} style Style for this feature.
+ * @param {module:ol/style/Style~Geometry|Array.<module:ol/style/Style~Geometry>|module:ol/style~StyleFunction} style Style for this feature.
```
Comment (tschaub) [inline src/ol/Feature.js]: @ahocevar - I'm guessing that this should be `module:ol/style/Style~Style`, right?

See #7961.

Comment (ahocevar) [reply src/ol/Feature.js]: Absolutely. Looks like that was too much search and replace.

File: src/ol/View.js [hunk 2]
```diff
@@ -24,6 +24,116 @@ import {createProjection, METERS_PER_UNIT} from './proj.js';
...
+ * between 0 and 1 representing the progress toward the destination state.
+ * @property {function(boolean)} [callback] Function called when the view is in
+ * its final position. The callback will be called with `true` if the animation
+ * series completed on its own or `false` if it was cancelled.
+ */
+
+
+/**
+ * @typedef {Object} ViewOptions
+ * @property {module:ol/coordinate~Coordinate~Coordinate} [center] The initial center for
```
Comment (tschaub) [inline src/ol/View.js]: Looks like an extra `Coordinate` here.  See #7964.

File: src/ol/Overlay.js [hunk 3]
```diff
@@ -11,6 +11,57 @@ import {listen, unlistenByKey} from './events.js';
...
+ * @typedef {Object} OverlayOptions
+ * @property {number|string} [id] Set the overlay id. The overlay id can be used
+ * with the {@link module:ol/Map~Map#getOverlayById} method.
+ * @property {Element} [element] The overlay element.
+ * @property {Array.<number>} [offset=[0, 0]] Offsets in pixels used when positioning
+ * the overlay. The first element in the
+ * array is the horizontal offset. A positive value shifts the overlay right.
+ * The second element in the array is the vertical offset. A positive value
+ * shifts the overlay down.
+ * @property {module:ol/coordinate~Coordinate~Coordinate} [position] The overlay position
```
Comment (tschaub) [inline src/ol/Overlay.js]: #7964 removes the extra `Coordinate`.

Comment (ahocevar) [reply src/ol/Overlay.js]: Thanks!

**Pull request merged**